### PR TITLE
Adding scale cluster command

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/gsctl/issues/93 https://github.com/giantswarm/giantswarm/issues/1461

### Preview:

```nohighlight
$ gsctl --api-endpoint=http://localhost:9000 scale cluster g1stg -w 5
Desired worker node count equals the current one.
No worker nodes have been added or removed.

$ gsctl --api-endpoint=http://localhost:9000 scale cluster g1stg -w 6
The cluster is being scaled up
Adding 1 worker to the cluster for a total of 6 workers.

$ gsctl --api-endpoint=http://localhost:9000 scale cluster g1stg -w 5
Do you really want to reduce the worker nodes for cluster 'g1stg' to 5? [y/n]: y
The cluster is being scaled down
Removing 1 worker from the cluster for a total of 5 workers.

$ gsctl scale cluster --help
Increase or reduce the number of worker nodes in a cluster.

Caution:

When reducing the number of nodes, the selection of the node(s) to be removed
is non-deterministic. Workloads on the worker nodes to be removed will be
terminated, data stored on the worker nodes will be lost. Make sure to remove
only as many nodes as your deployment architecture can handle in a resilient
way.

Examples:

  gsctl scale cluster c7t2o -w 12

  gsctl scale cluster c7t2o -w 12 --num-cpus 4

Usage:
  gsctl scale cluster [flags]

Flags:
      --force                If set, no confirmation is required when reducing the number of workers.
  -h, --help                 help for cluster
      --memory-gb float32    RAM per added worker node.
      --num-cpus int         Number of CPU cores per added worker node.
  -w, --num-workers int      Number of worker nodes to have after scaling.
      --storage-gb float32   Local storage size per added worker node.

Global Flags:
      --api-endpoint string   The URL base path, to access the API (default "https://api.giantswarm.io")
      --auth-token string     Authorization token to use
      --config-dir string     Configuration directory path to use (default "/Users/marian/.config/gsctl")
  -v, --verbose               Print more information
```

### TODO

- [x] Allow for setting config details of workers to be added (KVM only)
- [x] Test against working API
- [x] Change gsclientgen branch name to master once https://github.com/giantswarm/gsclientgen/pull/15 is merged

### CODE REVIEW

**Comment by calvix** [comment]
I find it weird that when deleting cluster I have to specify via `-c` and when scaling I don't have to do that.
I would vote for consistency

```
gsctl scale cluster -c c7t2o -w 12
```

**Comment by marians** [comment]
@calvix Me too. That will be fixed. https://github.com/giantswarm/gsctl/issues/94

**Comment by marians** [comment]
Just noticed that a unit test needs fixing. On it.

**Comment by calvix** [comment]
```
      --memory-gb float32    RAM per added worker node.
      --num-cpus int         Number of CPU cores per added worker node.
      --storage-gb float32   Local storage size per added worker node.
```
This is intresting, This look super complex from the operator size.
So if i scale from 4 workers to 8 and also increase CPU from 2 to 6.
This means we may have to reconfigure autoscaling group or maybe even create a new one? Hard to say but this may lead into serious downtime if handled badly.

Also we should distinguish between aws and kvm, becasue in aws you cant specify cpu or ram, you need to specify instance type.

**Comment by marians** [comment]
These flags have no effect on AWS.

See here for more details on how scaling works and what restrictions apply: https://github.com/giantswarm/api-spec/blob/scale-cluster-2/spec.yaml#L239-L349

**Comment by marians** [comment]
Unit test is fixed.

File: commands/scale_cluster.go [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+		PreRun: scaleClusterPreRunOutput,
+
+		// Run calls the business function and prints results and errors.
+		Run: scaleClusterRunOutput,
+	}
+)
+
+const (
+	scaleClusterActivityName string = "scale-cluster"
+	getClusterActivityName   string = "get-cluster"
```
Comment (kopiczko) [inline commands/scale_cluster.go]: nitpick: no need to specify `string` type here.

File: commands/scale_cluster.go [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+	if result.numWorkersToAdd > 0 {
+		// scaling up
+		fmt.Println(color.GreenString("The cluster is being scaled up"))
+		if result.numWorkersToAdd == 1 {
+			fmt.Printf("Adding 1 worker to the cluster for a total of %d workers.\n",
+				result.numWorkersAfter)
+		} else {
+			fmt.Printf("Adding %d workers to the cluster for a total of %d workers.\n",
+				result.numWorkersToAdd, result.numWorkersAfter)
+		}
```
Comment (kopiczko) [inline commands/scale_cluster.go]: I'd do it like

```
workerWord := "workers"
if result.numWorkersToAdd > 1 {
        workerWord = "worker"
}
fmt.Printf("Adding %d %s to the cluster for a total of %d workers.\n",
        result.numWorkersToAdd, workerWord, result.numWorkersAfter)
```

Or even create a closure:

```
workerWord := func(i int) string {
        if i == 1 {
                return "worker"
        }
        return "workers"
}

Comment (marians) [inline commands/scale_cluster.go]: Good inspiration!

**Comment by marians** [comment]
@kopiczko Since you reviewed this, I changed the API call logic a bit, to remove a superfluous GET call for cluster details after scaling. Also adapted the test again.

**Pull request merged**